app/jobs/ingest_job.rb

```diff
@@ -0,0 +1,17 @@
+class IngestJob < Hyrax::ApplicationJob
+  queue_as Hyrax.config.ingest_queue_name
+
+  after_perform do |job|
+    opts = job.arguments.reverse.detect { |x| x.is_a? Hash } || {}
+    wrapper = job.arguments.first
+    ContentNewVersionEventJob.perform_later(wrapper.file_set, wrapper.user) if opts[:notification]
```

Comment:
It wasn't more flexibility at all.  `ActiveJob` callbacks are *far* more flexible than our one-offs **and actually work** without logical errors.  We had `before` callbacks with no symmetric `after`.  We don't provide matching `around` callbacks.  Rails has already worked this stuff out to a vast degree, and already has symmetric and complete before/after/around hooks.  In the context of jobs, I don't see our one-off callbacks as ever preferable.  

But there are various approaches to this that I considered (some, even implemented) before this proposed one that might be useful to restate:
- separate jobs: `IngestJob` and `IngestJobWithNotification`.  The latter extends the former and unilaterally applies the `after_perform` block.  PRO: options fall away.  CON: would need a descendant job for each existing job.  If a second (or more) such runtime option emerged, it would become a nonsensical matrix of classes, like `IngestJobWithBellsWithNotification` and `IngestJobWithBellsWithWhistlesWithNotification`
- some refinement, where the `after_perform` block moves into a base class (or even the main `Hyrax::ActiveJob` class).  I would consider implementing this when a second job adopts the same pattern, not before.
- adding to `JobIoWrapper`: did not pursue this because it seemed distinct from the local filesystem problem the class intends to solve.  The *same* `JobIoWrapper` (uploaded file) might be expected to be passed to a series of jobs, some of them with notifications, some without.  We shouldn't affect the persisted representation of the wrapper just to reflect that.  Indeed we CANNOT, if a given job wants to invoke two separate `perform_later`s, it would be, e.g.: 
```
Job1.perform_later(wrapper)
wrapper.notification = false
wrapper.save
Job2.perform_later(wrapper)
```
But given that we don't know when the jobs process, i.e. when the persisted wrapper is retrieved from DB, this will allow `Job1` to retrieve the state we intended for `Job2`.